### ISSUE TITLE
IndexError: Cannot choose from an empty sequence when stack is empty

### ISSUE THREAD

**Issue by fridex** [thoth-station/adviser#1865]
**Describe the bug**

```
2021-05-12 08:38:09,407 thoth.adviser.run           ERROR: The resolution failed as an error was encountered: Cannot choose from an empty sequence
Traceback (most recent call last):
  File "/opt/app-root/src/thoth/adviser/run.py", line 75, in subprocess_run
    report: Union[DependencyMonkeyReport, Report] = resolver.resolve(
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1299, in resolve
    for state in self._do_resolve_states(with_devel=with_devel, user_stack_scoring=user_stack_scoring):
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1229, in _do_resolve_states
    for final_state in self._do_resolve_states_raw(
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1165, in _do_resolve_states_raw
    state, unresolved_package_tuple = self.predictor.run()
  File "/opt/app-root/src/thoth/adviser/predictors/td.py", line 139, in run
    unresolved_dependency_tuple = probable_state.get_random_unresolved_dependency(prefer_recent=True)
  File "/opt/app-root/src/thoth/adviser/state.py", line 219, in get_random_unresolved_dependency
    dependency_name = dependency_name or random.choice(list(self.unresolved_dependencies))
  File "/usr/lib64/python3.8/random.py", line 290, in choice
    raise IndexError('Cannot choose from an empty sequence') from None
IndexError: Cannot choose from an empty sequence
2021-05-12 08:38:09,433 thoth.adviser.run           ERROR: Child exited with exit code 512
2021-05-12 08:38:09,434 thoth.adviser.run           ERROR: Resolution was terminated based on errors encountered; see logs for more info - https://thoth-station.ninja/j/error_logs
```

**To Reproduce**
1. Create a Pipfile with a package that is known to be filtered (e.g. `enum34`)
2. Issue `thamos advise`
3. See the error

**Expected behavior**

Adviser should report back an empty stack.

**Comment by fridex** [comment]
Interestingly, the reporting is correct when running the same adviser version locally.

**Comment by fridex** [comment]
I was using wrong recommendation type, fix in #1867
